pkg/common/oras/authprovider/azure/azureworkloadidentity.go

```diff
@@ -205,6 +219,57 @@ func (d *WIAuthProvider) Provide(ctx context.Context, artifact string) (provider
 	return authConfig, nil
 }
 
+// parseEndpoints checks if the endpoints are valid for auth provider. If no
+// endpoints are provided, it defaults to the default ACR endpoint.
+// A valid endpoint is either a fully qualified domain name or a wildcard domain
+// name folloiwing RFC 1034.
+// Valid examples:
+// - *.example.com
+// - example.com
+//
+// Invalid examples:
+// - *
+// - example.*
+// - *example.com.*
+// - *.
+func parseEndpoints(endpoints []string) ([]string, error) {
+	if len(endpoints) == 0 {
+		return []string{defaultACREndpoint}, nil
+	}
+	for _, endpoint := range endpoints {
+		switch strings.Count(endpoint, "*") {
+		case 0:
+			continue
+		case 1:
+			if !strings.HasPrefix(endpoint, "*.") {
```

Comment:
we should update the docs so people are aware of this wildcard convention?

Reply:
good call, I added it in the chart readme and will have a more detailed doc in ratify-web.

Reply:
sounds good. It can be a follow up after this one. Also when a user provides value to the endpoints, the default `*.azurecr.io` is no longer automatically assumed. This is a subtle behavior and expectation should be clear that if user needs to provide all approriate values if endpoints specified